src/VaultSharp/V1/AuthMethods/CloudFoundry/Signature/CloudFoundrySignatureProvider.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Threading;
+using Org.BouncyCastle.Asn1;
+using Org.BouncyCastle.Asn1.Pkcs;
+using Org.BouncyCastle.Crypto.Digests;
+using Org.BouncyCastle.Crypto.Engines;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.Crypto.Signers;
+using Org.BouncyCastle.Utilities.IO.Pem;
+
+namespace VaultSharp.V1.AuthMethods.CloudFoundry.Signature
+{
+    public class CloudFoundrySignatureProvider
```

Comment:
can we get this out of VaultSharp and into the documentation as a helper? In general, i avoid file reading, locking etc. in the library and have the caller supply it in.

The docs can give pointers to the caller.